Add LLVM 20 package, stream LLVM 18 and 19 as mono sources.

javacruft opened the pull request:
Rename llvm package to llvm-19 and twiddle provides and package names to support version stream approach.

Add llvm-20 package based on llvm-19.

Add llvm-18 package based on llvm-19 and drop other packages that build the same binaries separately.

Drop provides for libunwind and lld packages for older LLVM versions; nice idea but its not worked out well.

Withdraw older llvm packages to remove these provides from the Wolfi archive.

javacruft committed:
llvm: refactor into version streamed package.

javacruft committed:
llvm-19: correct package name, update runtime deps for libclang-cpp

javacruft committed:
llvm-19: reorder vars/vars-transforms.

javacruft committed:
llvm-19: revert use of var-transforms in vars

javacruft committed:
llvm-20: add new package for release 20.x.

javacruft committed:
llvm: filter for version stream aligned tags.

javacruft committed:
llvm: drop provides for actual package name.

javacruft committed:
llvm: ensure backwards compatibility with llvm package.

javacruft committed:
llvm-20: drop clang-rename as no longer shipped upstream.

javacruft committed:
llvm-20: drop clang-pseudo binary testing and llvm-debuginfod-find --version

javacruft committed:
llvm-20: correct tag-filter for upstream versions.

javacruft committed:
llvm-{19,20}: correct naming of additional clang packages.

javacruft committed:
llvm-20: correct typo ldd->lld

javacruft committed:
llvm-{19,20}: fix provides and runtime for -dev packages.

javacruft committed:
bpftool: drop use of versioned llvm as this is picked by the installed clang version.

javacruft committed:
llvm-{19,20}: Tidy provides

Drop provides on LLVM subpackages for older versions; this approach
has not been successful; withdrawm older llvm package versions to keep
the archive tidy.

Use generated major version where possible to ease introduction of
new LLVM releases.

javacruft committed:
llvm-19: fix missed major-version usage.

javacruft committed:
Merge branch 'main' into llvm/llvm-20

javacruft committed:
llvm: add llvm-libunwind-18 as will be dropped from llvm.

javacruft committed:
llvm-libunwind-18: use full package versions for consistent installs.

javacruft committed:
llvm: create single LLVM 18 package

Drop individual LLVM 18 based packages and update LLVM 18 to align
to 19 and 20 package versions.

javacruft committed:
Merge branch 'main' into llvm/llvm-20

javacruft committed:
llvm20: bump version

javacruft committed:
llvm-18: quote version string.

javacruft committed:
llvm-18: set epoch to a version greater than all single sub-package packages.

javacruft committed:
llvm-18: drop install of c++ modules.json, not present for this release.

javacruft committed:
llvm-{18-20}: misc fixes

Ensure matched version of libcxxabi1 is installed with libcxx1.

Fix libcxx1 package install for llvm-18.

javacruft committed:
llvm-18: drop test for reduce-chunk-list, not build for this version.

javacruft committed:
llvm-18: drop test for clang-nvlink-wrapper, not built for this version.

javacruft committed:
postgis-17: pin clang to 19 as upstream build assumes this version.

javacruft committed:
bump epochs for test rebuild of packages with versionless clang.

javacruft committed:
Merge branch 'main' into llvm/llvm-20

javacruft committed:
Revert "bump epochs for test rebuild of packages with versionless clang."

This reverts commit ba2425003d14dd77c3ef96dcb536d11789c2fafd.

javacruft committed:
Pin various postgresql related packages to clang-19

postgresql uses clang-19 to build so related packages that use
the postgresql dev packages need to be version aligned.

javacruft commented:
Versionless clang test rebuild in wolfi:

✅ bpftool
✅ bzip3
✅ chromium
✅ deno
✅ falco-no-driver
✅ gn
✅ libbpf
✅ libmamba
✅ linkerd2
✅ linkerd2-proxy
✅ llvm-libcxx-15
✅ llvm-libcxx-16
✅ llvm-libcxx-17
✅ llvm-libunwind-15
✅ llvm-libunwind-16
✅ llvm-libunwind-17
✅ neon
✅ oath-toolkit
✅ opencv
✅ pax-utils
✅ postgis-17
✅ rpm-sequoia
✅ rust-bindgen
✅ svt-av1
✅ xcover
✅ ztunnel-1.26

Failed due to mismatched clang version with postgresql-dev:

❌ pgaudit-17
❌ pg-failover-slots-16
❌ pgvector-16
❌ pgvector-17

included on this PR for resolution.

javacruft committed:
Revert "Pin various postgresql related packages to clang-19"

This reverts commit 25eb08ff8a002d3eb3efa58d9666902269589857.

javacruft committed:
Merge branch 'main' into llvm/llvm-20

javacruft committed:
Drop changes to postgis-17; covered elsewhere

javacruft committed:
Merge branch 'main' into llvm/llvm-20

javacruft committed:
llvm-20: bump upstream release.

javacruft committed:
Merge branch 'main' into llvm/llvm-20

javacruft committed:
llvm-{18-20}: Drop openssf-compiler-options from BD's

EyeCantCU reviewed (changes requested):
Amazing work! Thank you for bringing up LLVM 20! I just have a few comments

javacruft committed:
llvm: use versioned package names for internal package dependencies.

javacruft committed:
llvm-{18,19}: drop provides

Drop unversioned streamed provides; this allows co-installability
between LLVM versions, with just the most recent providing versionless
provides packages.

javacruft commented:
@EyeCantCU I believe I've address all of your feedback - thanks for the review!

EyeCantCU commented:
Just needs one final +1 from OS :)

javacruft commented:
On Sat, 31 May 2025 at 15:02, RJ Trujillo ***@***.***> wrote:

> *EyeCantCU* left a comment (wolfi-dev/os#52810)
> <https://github.com/wolfi-dev/os/pull/52810#issuecomment-2925224017>
>
> Just needs one final +1 from OS :)
>
Great - thanks for the reviews! Aiming to get this merged on Monday AM so
will find a European TZ OS team member to provide the final approval.

—
> Reply to this email directly, view it on GitHub
> <https://github.com/wolfi-dev/os/pull/52810#issuecomment-2925224017>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADXJOGPWSAMXVYFJQGHDL33BGY6RAVCNFSM6AAAAAB4WEIEV6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSMRVGIZDIMBRG4>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>
